### DISCUSSION TITLE
Searching for a substring with a newline using the "contains" function

### DISCUSSION THREAD

**Discussion by bryangingechen** [community/community#25134] [Actions]
<p>I have something like the following in a Github Actions workflow:</p>
<pre><code>on:
  issue_comment:
    types: [created]

jobs:
  job:
    name: My job
    if: startsWith(github.event.comment.body, 'foo') || contains(github.event.comment.body, '\nfoo')
</code></pre>
<p>The intention is that the job only runs if a comment is posted that contains “foo” at the start of a line. However, the second clause never seems to fire. I suspect this is because ‘\n’ is not being interpreted as a newline character. Are special characters in literals escaped in a different way?</p>
<p>How should I rewrite my workflow so that the job will run when I post a comment like:</p>
<pre><code>test
foo
</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/bryangingechen">@bryangingechen</a> ,</p>
<p>You can set to return the value of  <strong>github.event.comment.body</strong> as <strong>JSON-type</strong>  via the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#tojson" rel="nofollow noopener"><strong>toJson</strong></a> function. In the JSON-type, the <strong>line-break</strong> will be converted to ’ <strong>\r\n</strong>’.</p>
<pre><code>if: startsWith(github.event.comment.body, 'foo') || contains(toJson(github.event.comment.body), '\r\nfoo')
</code></pre>

**Comment by bryangingechen** [comment]
<p>Thank you, that works perfectly!</p>
